Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.mm

```diff
@@ -96,13 +99,20 @@
       // Cancel the request since the user changed while the request was
       // outstanding so the response is likely for a previous user (which
       // user, we can't be sure).
-      completion(Token::Invalid(), FirestoreErrorCode::Aborted,
-                 "getToken aborted due to user change.");
+      completion(util::Status(FirestoreErrorCode::Aborted,
+                              "getToken aborted due to user change."));
     } else {
-      completion(
-          Token{util::MakeStringView(token), contents->current_user},
-          error == nil ? FirestoreErrorCode::Ok : error.code,
-          error == nil ? "" : util::MakeStringView(error.localizedDescription));
+      if (error == nil) {
+        completion(Token{util::MakeStringView(token), contents->current_user});
```

Comment:
This creates a semi-broken string view, wherein the contents are actually the `nullptr`. If `token` is `nil`, [`MakeStringView` doesn't do anything special to ensure the string view is the empty string](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/util/string_apple.h#L57).

Reply:
What does `[str lengthOfBytesUsingEncoding:NSUTF8StringEncoding]` return if `str` is `nil`?

Reply:
Objective-C has a notion of nil dispatch. Essentially if you call a method on `nil` the result will be some zero value appropriate to the type: a nil pointer for NSObject-derived pointer types (including NSString), nullptr for unmanaged pointer types, zero for integral types, etc.

I believe MakeStringView would construct an `absl::string_view{nullptr, 0}` if passed a nil token.

Reply:
Thanks for the explanation. I _think_ it's a valid way to create an empty `string_view` ([doc](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/string_view.h#L113-L119)), unless some code relies on the assumption that empty string view is `== ""`.